File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = 0;
        int curHealth = health;
        int attackIdx = 0;
        int cnt = 0;
        for(int time=1;time<=attacks[attacks.length-1][0];time++){
            if(time == attacks[attackIdx][0]){
                curHealth -= attacks[attackIdx][1];
                attackIdx++;
                cnt = 0;
            } else{
                curHealth = Math.min(health, curHealth + bandage[1]);
                cnt++;
                
                if(cnt == bandage[0]){
                    curHealth = Math.min(health, curHealth + bandage[2]);
                    cnt = 0;
                }
            }
            
            if(curHealth <= 0)
                return -1;
        }
        return curHealth;
    }
}

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    attack_idx = 0
    cnt = 0
    cur_health = health
    for time in range(1,attacks[-1][0]+1):
        if time==attacks[attack_idx][0]:
            cur_health -= attacks[attack_idx][1]
            attack_idx += 1
            cnt = 0
        else:
            cur_health = min(health, cur_health+bandage[1])
            cnt += 1
            if cnt==bandage[0]:
                cur_health = min(health, cur_health+bandage[2])
                cnt = 0
        if cur_health<=0:
            return -1
    return cur_health